src/components/FactoryBlock/Block.jsx

```diff
@@ -3,15 +3,28 @@ import { Card } from '@material-ui/core'
 import connector from './BlockConnector'
 import { machineByType, Empty } from '../../utils/machineUtils'
 import './Block.css'
+import { SELECTION, DELETE } from '../../utils/editionUtils'
+
+const actionHandler = ({ action, position, deleteDispatch }) => {
+  if (action === DELETE) {
+    deleteDispatch(position)
+  }
+}
 
 class Block extends Component {
   onClick = () => {
-    const { position, machineSelected, updateBlock, node } = this.props
+    const { position, machineSelected, actionSelected, updateBlock, node, deleteBlock } = this.props
+
+    // @TODO maybe this check could be in another place
+    if (!machineSelected && actionSelected === SELECTION) return
 
     if (machineSelected && node.type === Empty) {
-      // TODO: maybe this check should be in another place
       updateBlock(position, machineSelected.name)
     }
+
+    if (actionSelected !== SELECTION && node.type !== Empty) {
```

Comment:
I think that this checks doesn't should be here, i was thinking resolve this checks in #71

Reply:
well I think we should stop trying to add fixes in development issues, I was trying to separate that logic in a function but I think all of us have to decide what could be the best way to do it.
In other word, stop doing hotfixes inside feature issues.